python/meep.i

```diff
@@ -152,6 +153,36 @@ static int py_list_to_gobj_list(PyObject *po, geometric_object_list *l);
     delete[] $1.items;
 }
 
+// Typemap suite for sources
+
+%typecheck(SWIG_TYPECHECK_POINTER) const meep::src_time & {
+    int py_source_time = PyObject_IsInstance($input, py_source_time_object());
+    int swig_src_time = PyObject_IsInstance($input, py_meep_src_time_object());
+
+    $1 = py_source_time || swig_src_time;
+}
+
+%typemap(in) const meep::src_time & {
+    PyObject *swig_obj = NULL;
+    void *tmp_ptr = 0;
+    int tmp_res = 0;
+
+    if(PyObject_IsInstance($input, py_source_time_object())) {
+        swig_obj = PyObject_GetAttrString($input, "swigobj");
```

Comment:
This returns a new reference, so you need to `Py_DECREF(swig_obj)` after the `tmp_res = ...`

However, `swig_obj = $input` does *not* create a new reference, so you will need a `Py_INCREF(swig_obj)` there to balance out the decref.